cocos/3d/CCSkybox.cpp

```diff
@@ -152,18 +152,32 @@ void Skybox::draw(Renderer* renderer, const Mat4& transform, uint32_t flags)
 
 void Skybox::onDraw(const Mat4& transform, uint32_t flags)
 {
-    Mat4 trans(transform);
-    const cocos2d::Vec3 pos(Camera::getVisitingCamera()->getPosition3D());
-    trans.m[12] = pos.x;
-    trans.m[13] = pos.y;
-    trans.m[14] = pos.z;
-
+    auto camera = Camera::getVisitingCamera();
+    
+    /*
+     At beginning, we have a regular skybox at origin point.
+     To render the skybox, we should keep camera at the center of skybox.
+     So we need a translate matrix, which can transform origin point to camera pos.
+     Camera's node to word transform matrix don't match our requement,
+        because it maybe contain the scale, rotate, reflact... effects, which isn't need.
+     First, we transform origin point to camera position by camera's node to world matrix.
+     Second, we create a translate matrix with the origin point's world position.
+     */
+    Vec3 cameraPosInNode(0, 0, 0);
+    Vec3 cameraPosInWorld;
+    Mat4 cameraModelMat = camera->getNodeToWorldTransform();
```

Comment:
` Mat4 cameraModelMat = camera->getNodeToWorldTransform();` try to use `auto&` to avoid extra copy?